app/controllers/drafts/ViewAllDraftMovementsController.scala

```diff
@@ -0,0 +1,115 @@
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class ViewAllDraftMovementsController @Inject()(mcc: MessagesControllerComponents,
+                                                getDraftMovementsConnector: GetDraftMovementsConnector,
+                                                view: ViewAllDraftMovementsView,
+                                                getExciseProductCodesConnector: GetExciseProductCodesConnector,
+                                                errorHandler: ErrorHandler,
+                                                val auth: AuthAction,
+                                                val getData: DataRetrievalAction,
+                                                val betaAllowList: BetaAllowListAction,
+                                                formProvider: ViewAllDraftMovementsFormProvider,
+                                                paginationHelper: DraftMovementsPaginationHelper
+                                          )(implicit val executionContext: ExecutionContext, appConfig: AppConfig)
+  extends FrontendController(mcc)
+    with AuthActionHelper with I18nSupport with BetaChecks {
+
+  def onPageLoad(ern: String, searchOptions: GetDraftMovementsSearchOptions): Action[AnyContent] = {
+    authorisedDataRequestAsync(ern, searchMovementsBetaGuard(ern)) { implicit request =>
+      renderView(Ok, ern, searchOptions, formProvider())
+    }
+  }
+
+  def onSubmit(ern: String, searchOptions: GetDraftMovementsSearchOptions): Action[AnyContent] = {
+    authorisedDataRequestAsync(ern, searchMovementsBetaGuard(ern)) { implicit request =>
+      formProvider().bindFromRequest().fold(
+        formWithErrors => renderView(BadRequest, ern, searchOptions, formWithErrors),
+        value => Future(Redirect(routes.ViewAllDraftMovementsController.onPageLoad(ern, value)))
+      )
+    }
+  }
+
+  private def renderView(
+                          status: Status,
+                          ern: String,
+                          searchOptions: GetDraftMovementsSearchOptions,
+                          form: Form[GetDraftMovementsSearchOptions]
+                        )(implicit request: DataRequest[_]): Future[Result] = {
+
+    val result: EitherT[Future, ErrorResponse, Result] = for {
+      draftMovements <- EitherT(getDraftMovementsConnector.getDraftMovements(ern, Some(searchOptions)).map {
+        case Left(NotFoundError) => Right(GetDraftMovementsResponse(0, Seq.empty))
+        case value => value
+      })
+      exciseCodes <- EitherT(getExciseProductCodesConnector.getExciseProductCodes())
+      exciseCodesWithDefault = GetDraftMovementsSearchOptions.CHOOSE_PRODUCT_CODE +: exciseCodes
+    } yield {
+
+      val pageCount: Int = calculatePageCount(draftMovements)
+
+      if (searchOptions.index <= 0 || searchOptions.index > pageCount) {
+        Redirect(routes.ViewAllDraftMovementsController.onPageLoad(ern, GetDraftMovementsSearchOptions()))
```

Comment:
What would be better here, would be to retain all the other search parameters. So, copy the `searchOptions` and then set the `index` to `1`. Keeping everything else unchanged.